exams/pearson_test.py

```diff
@@ -0,0 +1,191 @@
+
+        addr1_field_mapper = get_field_mapper(get_addr1)
+
+        assert addr1_field_mapper == get_addr1
+        assert addr1_field_mapper(profile) == profile.address1
+
+        with self.assertRaises(TypeError):
+            get_field_mapper([])
+
+    def test_profile_to_ccd_row_invalid_country(self):  # pylint: disable=no-self-use
+        """
+        A profile with an invalid country code should raise an InvalidProfileDataException
+        """
+        with mute_signals(post_save):
+            profile = ProfileFactory.create()
+        profile.country = 'XXXX'
+        with self.assertRaises(InvalidProfileDataException):
+            profile_country_to_alpha3(profile)
+
+    def test_write_profiles_ccd_no_profiles(self):  # pylint: disable=no-self-use
+        """
+        Tests write_profiles_ccd against an empty set of profiles
+        """
+        file = io.StringIO()
+
+        ccd_writer(file, [])
+
+        lines = file.getvalue().splitlines()
+        header = lines[0].split('\t')
+
+        assert len(lines) == 1
+        assert header == [
+            'ClientCandidateId',
+            'FirstName',
+            'LastName',
+            'Email',
+            'Address1',
+            'Address2',
+            'Address3',
+            'City',
+            'State',
+            'PostalCode',
+            'Country',
+            'Phone',
+            'PhoneCountryCode',
+            'LastUpdate',
+        ]
+
+    def test_ccd_writer(self):  # pylint: disable=no-self-use
+        """
+        Tests ccd_writer against a set of profiles
+        """
+        file = io.StringIO()
+
+        exam_profiles = []
+        with mute_signals(post_save):
+            exam_profiles.append(ExamProfileFactory.create(profile=ProfileFactory.create()))
+            exam_profiles.append(ExamProfileFactory.create(profile=ProfileFactory.create(address3='Room B345')))
+            exam_profiles.append(ExamProfileFactory.create(profile=ProfileFactory.create(address2=None)))
+
+        ccd_writer(file, exam_profiles)
+
+        lines = file.getvalue().splitlines()
+
+        rows = lines[1:]
+
+        for idx, exam_profile in enumerate(exam_profiles):
+            profile = exam_profile.profile
+            row = rows[idx]
+            country = pycountry.countries.get(alpha_2=profile.country)
+            assert row == ('\t'.join(map(str, [
+                profile.student_id,
+                profile.romanized_first_name,
+                profile.romanized_last_name,
+                profile.user.email,
+                profile.address1,
+                profile.address2 or '',
+                profile.address3 or '',
+                profile.city,
+                profile.state_or_territory,
+                profile.postal_code,
+                country.alpha_3,
+                profile.phone_number,
+                profile.phone_country_code,
+                profile.updated_on.strftime(PEARSON_DATETIME_FORMAT),
+            ])))
```

Comment:
This test feels a little too high-level to me. I understand the desire to not repeat yourself, but it's hard to see what the expected inputs and expected outputs actually are. Can we add a really concrete test? Something like:

```python
profile = ExamProfileFactory.create(
    student_id=37,
    romanized_first_name="Carl",
    romanized_last_name="Smith",
    user__email="abc@example.com",
    address1="123 Main Street",
    city="Springfield",
    state_or_territory="MA",
    postal_code="98765",
    phone_number="999-999-9999",
    phone_country_code="1",
    updated_on=datetime(2016, 12, 15, 10, 15, 0),
)
# ...
assert exported_row == (
    "37\tCarl\tSmith\tabc@example.com\t"
    "123 Main Street\t\t\tSpringfield\tMA\t"
    "98765\tUSA\t"
    "999-999-9999\t1\t"
    "2016/12/15 10:15:00\n"
)
```

That way, it becomes much easier to see what you actually expect the input and output formats to look like.